src/components/views/rooms/MessageComposer.js

```diff
@@ -192,9 +192,29 @@ module.exports = React.createClass({
     },
 
     onAction: function(payload) {
+        var textarea = this.refs.textarea;
         switch (payload.action) {
             case 'focus_composer':
-                this.refs.textarea.focus();
+                textarea.focus();
+                break;
+            case 'insert_displayname':
+                if (textarea.value) {
+                    var left = textarea.value.substring(0, textarea.selectionStart);
+                    var right = textarea.value.substring(textarea.selectionEnd);
+                    if (right && right.length) {
```

Comment:
`left` will always have a value (even if it's a null string), as it's the substring from index 0 to a string of known finite length.  this is the only place we dereference `right`.  Ideally we'd be put checks everywhere but it'd clutter it up quite a lot...